src/SFA.DAS.EmployerIncentives.Data/ApprenticeshipIncentives/ApprenticeshipIncentiveDataRepository.cs

```diff
@@ -151,6 +129,34 @@ public async Task Delete(ApprenticeshipIncentiveModel apprenticeshipIncentive)
             }
         }
 
+        private async Task<ApprenticeshipIncentiveModel> Get(Expression<Func<ApprenticeshipIncentive, bool>> predicate)
+        {
+            var apprenticeshipIncentive = await _dbContext.ApprenticeshipIncentives.FirstOrDefaultAsync(predicate);
+
+            if (apprenticeshipIncentive != null)
+            {
+                await PopulateApprenticeshipIncentive(apprenticeshipIncentive);
+                return apprenticeshipIncentive?.Map(_dbContext.CollectionPeriods.AsEnumerable());
+            }
+
+            return null;
+        }
+
+        private async Task PopulateApprenticeshipIncentive(ApprenticeshipIncentive apprenticeshipIncentive)
+        {
+            apprenticeshipIncentive.PendingPayments = await _dbContext.PendingPayments.Where(x => x.ApprenticeshipIncentiveId == apprenticeshipIncentive.Id).ToListAsync();
+            foreach (var pendingPayment in apprenticeshipIncentive.PendingPayments)
+            {
+                pendingPayment.ValidationResults = await _dbContext.PendingPaymentValidationResults.Where(x => x.PendingPaymentId == pendingPayment.Id).ToListAsync();
+            }
+            apprenticeshipIncentive.Payments = await _dbContext.Payments.Where(x => x.ApprenticeshipIncentiveId == apprenticeshipIncentive.Id).ToListAsync();
```

Comment:
Would there be any harm in doing these gets concurrently? As long as multiple active record sets is enabled it should be quicker?

Reply:
@MichaelYoung1981 it will need DBContext Factory, its still doable not a big deal
https://stackoverflow.com/questions/41749896/ef-6-how-to-correctly-perform-parallel-queries

Reply:
Have tried that, unfortunately the queries can't be run in a separate data context as it prevents the changes from being tracked and persisted in the Unit of Work